docs/QueryOptimizer.md

```diff
@@ -0,0 +1,183 @@
+
```

Comment:
Should we use a lower case pattern for page docs here? e.g. `docs/query-optimizer.md` instead? I think it looks cleaner

Reply:
Agree.